### PR TITLE
Add PEP8 test to Travis CI

### PR DESCRIPTION
**Pull request by pllim**
Add PEP8 test but allow it to fail. Similar to astropy/astropy#5107.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -92,6 +97,11 @@ matrix:
         - python: 3.5
           env: NUMPY_VERSION=prerelease
 
+    allow_failures:
+        # Do a PEP8 test (allow to fail unless your code completely compliant)
+        - os: linux
+          env: MAIN_CMD='pep8 packagename --count' SETUP_CMD=''
```
Comment (bsipocz) [inline .travis.yml]: ``pep8`` is renames to ``pycodestyle``, please us the latter.

Comment (pllim) [inline .travis.yml]: Good point. Do I also have to make that change in `setup.cfg`? I don't see it in astropy/astropy#5107

Comment (bsipocz) [inline .travis.yml]: Ideally yes, but last time I checked it was not yet supported, need to check whether they had a new release since.

**Comment by pllim** [comment]
Hmm I don't see "allowed failures" section in the Travis CI of this PR. Did I do something wrong?

**Comment by bsipocz** [comment]
@pllim - Because you didn't include the pycodestyle build into the matrix. One has to repeat a build under allow failures to actually allow them to fail.

**Comment by bsipocz** [comment]
This is non controversial and definitely an improvement so I'm merging. Thanks @pllim.

**Pull request merged**